lib/role_model/class_methods.rb

```diff
@@ -41,21 +41,20 @@ def roles(*roles)
       opts = roles.last.is_a?(Hash) ? roles.pop : {}
       self.valid_roles = roles.flatten.map(&:to_sym)
       unless (opts[:dynamic] == false)
-        self.define_dynamic_queries(self.valid_roles)
+        self.define_dynamic_queries(self.valid_roles, opts[:prefix])
       end
     end
-    
-    # Defines dynamic queries for :role
-    #   #is_<:role>?
+
+    # Defines dynamic queries for :role. Providing a prefix is optional.
     #   #<:role>?
+    #   #<:role>=(true|false)
     #
-    # Defines new methods which call #is?(:role)
-    def define_dynamic_queries(roles)
+    # Defines new methods which call #is?(:role) and roles.(add|remove)
+    def define_dynamic_queries(roles, prefix=nil)
```

Comment:
one could make the argument that nil is appropriate as well since there is actually "no prefix". In other words, even if I didn't do `"#{prefix}"`, a programmer should still check that it is present. 

I'm basically saying "this code works **without** a prefix when I do =nil. When I do ="", it says, I **need** a value there because I didn't plan for the absence of it.

I was very specific about my use of nil.

If you still feel strongly about it, I will change, just let me know.